tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -51,6 +65,14 @@ TRTDeviceAllocator::TRTDeviceAllocator(tensorflow::Allocator* allocator)
 
 void TRTDeviceAllocator::free(void* memory) {
   VLOG(2) << "Deallocating @ " << memory;
+  // allocated memory adjusted for alignment, restore the original pointer
+  if (mem_map.count(memory) != 0) {
```

Comment:
It would be more efficient to implement this as:
```
auto alloc_mem_it = mem_map.find(memory);
if (alloc_mem_it != mem_map.end()) {
  memory = *alloc_mem_it;
  mem_map.erase(alloc_mem_it);   // O(1) complexity.
}
CHECK(mem_pool.erase(memory) != 0);
```